rotkehlchen/exchanges/gemini.py

```diff
@@ -507,15 +543,19 @@ def query_online_deposits_withdrawals(
             end_ts=end_ts,
         )
         movements = []
+        history_events: list[HistoryEvent] = []
```

Comment:
So I see what you are doing here but there is multiple small problems which I believe are fixable.

1. This will only fire when you navigate to exchange withdrawals page or rotki periodically decides to try and refresh them.
2. It mixes the processing and query of history events and of asset movements. I guess the reason is since they both come from the same endpoint. So it makes sense to not query it twice but then it would need to be reorganized.
3. To reorganize them perhaps move the logic of this into a `_query_transfers` function much like we do in coinbase for `_query_transactions`. And instead of returning the movements, inside the function just like coinbase write them to the DB.

Now some problems will come out of that. You will:

1. Need to specify in the docstring that even though start_ts/end_ts are for asset movements they may also include some history events.
2. We need whenever this patch is released to think of how to clean Gemini data. As we need to reset this query to the start. And by doing so duplications will happen. So would need to probably purge and requery all gemini data.